### PR TITLE
add AssistNow Online and Offline messages for GPS, GLO, GAL

### PR DESCRIPTION
**Pull request by Kriechi**
## Description

This PR adds UBX messages for AssistNow Online and AssistNow Offline messages provided by the u-blox services.

## Testing

There is one test failing now, but I think this is expected. I'm not sure though why there is a test with a valid message which was never implemented. Please advise if I should delete the test (because the message is now implemented) or rewrite the test to produce another parse error: `test_exceptions.py", line 280, in testParseMode2`

## Checklist:

- [X] My code follows the style guidelines of this project (see `CONTRIBUTING.MD`).
- [X] I have performed a self-review of my own code.
- [X] I have commented my code, particularly in hard-to-understand areas.
- [X] I have made corresponding changes to the documentation.
- [X] (*if appropriate*) I have added test cases to the `tests/test_*.py` unittest suite to maintain >= 99% code coverage.
- [X] I have tested my code against the full `tests/test_*.py` unittest suite.
- [X] My changes generate no new warnings.
- [X] Any dependent changes have been merged and published in downstream modules.
- [X] I understand and acknowledge that the code will be published under a BSD 3-Clause license.

### CODE REVIEW

File: pyubx2/ubxtypes_core.py [hunk 1]
```diff
@@ -231,40 +233,40 @@
     # These need special handling as MSGIDs alone are not unique;
     # message identity is determined by 'type' attribute in payload
     # ***************************************************************
-    b"\x13\x60\x01": "MGA-ACK-DATA0",
```
Comment (Kriechi) [inline pyubx2/ubxtypes_core.py:234]: To provide better readability, I've sorted this whole block to make the IDs be in ascending order.
No other change made here.

Comment (semuadmin) [reply pyubx2/ubxtypes_core.py:234]: Hi - appreciate the contribution and will review. Not sure I agree that sorting the block by msg/id necessarily aids readability - the blocks are currently sorted in alphabetical order of message name ("MGA-ACK-DATA0" etc.) and sorting this block differently creates unnecessary churn IMHO. I'd prefer to limit the delta to just the messages you've added or amended and to state explicitly in the PR preamble which specific messages these are.

**Review by semuadmin — COMMENTED** [review]
Hi - appreciate the contribution and will review. Not sure I agree that sorting the block by msg/id necessarily aids readability - the blocks are currently sorted in alphabetical order of message name ("MGA-ACK-DATA0" etc.) and sorting this block differently creates unnecessary churn IMHO. I'd prefer to limit the delta to just the messages you've added or amended and to state explicitly in the PR preamble which specific messages these are.

Also, can you confirm which u-blox specification you've referenced here which states that these are OUTPUT message types (i.e. sent from the receiver)? I'll happily stand corrected on this, but to the best of my knowledge, many (if not all) of the messages you've added in ubxtypes_get.py (i.e. the OUTPUT message definitions) are actually INPUT messages (i.e. messages sent TO the receiver from tu-blox's Assist Now facility) which should be defined in ubxtypes_set.py (which they already are).

Take MGA-ANO for example - in the documentation for, say, M9N receivers this is stated to be INPUT only:
ubx-mga-ano

**Comment by Kriechi** [comment]
On the sorting and readability: I've reverted the change.

Indeed, there might be some confusion on my side. What I'm essentially trying to do is to parse an AssistNow Online or Offline payload. So I've downloaded them from the ublox service and stored them into a file. I guess I never checked the `ubxtypes_set.py` file, as I was following one of the examples to read UBX messages from a file stream:
```python
f = open("file_online.bin", "rb")
ubr = UBXReader(f, ubxonly=True)

for raw_data, parsed_data in ubr:
    print(parsed_data)
```

**Comment by semuadmin** [comment]
> Indeed, there might be some confusion on my side. What I'm essentially trying to do is to parse an AssistNow Online or Offline payload. So I've downloaded them from the ublox service and stored them into a file. 

So, for example, if you take a look at the test file `/test/test_assistnow.py`, you'll see this is exactly what I'm doing here for a range of AssistNow ephemera and almanac payloads.

By default, the pyubx2 reader/parser assumes you're trying to parse an _Output_ message (i.e. one output by the receiver) as this is the most common use case. However, if you want to construct or parse an _Input_ message, such as one of the MGA payloads created by Assist Now facility, pyubx2 does support this - you simply need to set the `msgmode` parameter to SET (0x01) instead of the default GET (0x00).

So if, for example, you want to construct an MGA-GPS-EPH message from a raw AssistNow ephemera payload, with a view to sending it to the receiver (this example uses a real-world AssistNow payload): 

>>> from pyubx2 import UBXMessage, SET
>>> msg = UBXMessage('MGA', 'MGA-GPS-EPH', **SET**, payload=b'\x01\x00\x01\x00\x00\x01\x00\x0B\x4F\x00A4\x1F\x00\x00\xDB\xFF\x50\x5C\x1A\x00\x3A\x03\x4F\x2D\x19\x5867\x9E\x80\x02\xE4\x0D\x8D\x2E\x30\x05\xA2\x8E\x0D\xA1\xA4\x1F46\x00\x69\x76\xA5\xF8\xB9\xFF\xC7\x20\x14\x1D\x09\x28\xF4\xD3AA\x21\x2B\xAA\xFF\xFF\xAA\x04\x00\x00\x20\xF1')
>>> print(msg)
<UBX(MGA-GPS-EPH, type=1, version=0, svId=1, reserved1=0, fitInterval=0, uraIndex=1, svHealth=0, tgd=11, iodc=79, toc=13377, reserved2=31, af2=0, af1=-9472, af0=442257663, crs=14848, deltaN=20227, m0=911743277, cuc=-25033, cus=640, e=780996068, sqrtA=2392982832, toe=41229, cic=8100, omega0=1761621556, cis=-23178, crc=-17928, i0=337692671, omega=-198702819, omegaDot=557924819, idot=-21973, reserved3=78315519)>

If you want to parse an existing MGA-GPS-EPH message:

>>> from pyubx2 import UBXReader, SET
>>> msg = UBXReader.parse(b'\xb5b\x13\x00J\x00\x01\x00\x01\x00\x00\x01\x00\x0bO\x00A4\x1f\x00\x00\xdb\xffP\\\x1a\x00:\x03O-\x19X67\x9e\x80\x02\xe4\r\x8d.0\x05\xa2\x8e\r\xa1\xa4\x1f46\x00iv\xa5\xf8\xb9\xff\xc7 \x14\x1d\t(\xf4\xd3AA!+\xaa\xff\xff\xaa\x04\x00\x00 \xf1\n\xe0', **msgmode=SET**)
>>> print(msg)
<UBX(MGA-GPS-EPH, type=1, version=0, svId=1, reserved1=0, fitInterval=0, uraIndex=1, svHealth=0, tgd=11, iodc=79, toc=13377, reserved2=31, af2=0, af1=-9472, af0=442257663, crs=14848, deltaN=20227, m0=911743277, cuc=-25033, cus=640, e=780996068, sqrtA=2392982832, toe=41229, cic=8100, omega0=1761621556, cis=-23178, crc=-17928, i0=337692671, omega=-198702819, omegaDot=557924819, idot=-21973, reserved3=78315519)>

So I think pyubx2 already supports what you need. Hope this helps, but if you have any further queries, just reply to this message.

**Pull request closed without merging**